src/linear_algebra/dual.lean

```diff
@@ -206,3 +209,122 @@ def eval_equiv (h : dim K V < omega) : V ≃ₗ[K] dual K (dual K V) :=
 linear_equiv.of_bijective (eval K V) eval_ker (eval_range h)
 
 end vector_space
+
+section dual_pair
+
+open vector_space module module.dual linear_map function
+
+universes u v w
+variables {K : Type u} {V : Type v} {ι : Type w} [decidable_eq ι]
+variables [discrete_field K] [add_comm_group V] [vector_space K V]
+          [decidable_eq V] [decidable_eq (ι → V)] [decidable_eq $ dual K V]
+
+local notation `V'` := dual K V
+
+/-- `e` and `ε` have characteristic properties of a basis and its dual -/
+structure dual_pair (e : ι → V) (ε : ι → V') :=
+(eval : ∀ i j : ι, ε i (e j) = if i = j then 1 else 0)
+(total : ∀ {v : V}, (∀ i, ε i v = 0) → v = 0)
+[finite : ∀ v : V, fintype {i | ε i v ≠ 0}]
+
+end dual_pair
+
+namespace dual_pair
+
+local attribute [instance, priority 1] classical.prop_decidable
+
+open vector_space module module.dual linear_map function
+
+universes u v w
+variables {K : Type u} {V : Type v} {ι : Type w} [dι : decidable_eq ι]
+variables [discrete_field K] [add_comm_group V] [vector_space K V]
+          [decidable_eq V] [dιv : decidable_eq (ι → V)] [decidable_eq $ dual K V]
```

Comment:
This is still way more finicky than I would like. Short answer, no, they're not necessary. But it's nontrivial to remove them. I've done it (see the coming commit), but it took some trial and error, and I expect it will have an effect on the sensitivity proof (which I haven't checked against these changes yet).

The fact that they were unnecessary was also not caught by `sanity_check`; they're unnecessary, but not unused.

Reply:
Good news: removing these had no effect on the sensitivity proof.